### ISSUE TITLE
Test to submit a webextension on AMO

### ISSUE THREAD

**Issue by krupa** [mozilla/Addon-Tests#820]
## test webextensions:
https://github.com/mdn/webextensions-examples

## steps to reproduce:
1. Make sure your webextension has a unique id in the manifest.json
2. Load https://addons-dev.allizom.org/en-US/developers/addon/submit/2
3. Submit your test webextension
4. After validation, click Continue and complete submission

## expected behavior:
Submission is completed and the webextension has the status set to pending review.

**Comment by Edely** [comment]
Hi. I would like to work on this task.

**Comment by davehunt** [comment]
That's great @Edely! You'll need to create an account on https://addons-dev.allizom.org/ and get API credentials from https://addons-dev.allizom.org/en-US/developers/addon/api/key/. Once you have these let me know what email address you used to sign up and I'll make sure your account has the permission to create users via the API. You'll then be able to write a test that uses the `logged_in` fixture to submit a web extension. In the meantime I suggest trying the steps manually, and perhaps reading up a little on web extensions.

**Comment by Edely** [comment]
Hi, I created an account on addons-dev.allizom.org, but I am having a little trouble to get the credentials api. When I accept the Firefox Add-on Distribution Agreement I keep reloading this same page of agreement. However I did the same procedure on addons.allizom.org and I got the credentials without problems.

**Comment by Edely** [comment]
In fact, I could generate the credentials.

**Comment by davehunt** [comment]
> In fact, I could generate the credentials.

So are you unblocked and able to proceed?

**Comment by Edely** [comment]
Yes.

**Comment by davehunt** [comment]
> Yes.

That's great! Let me know what e-mail address you're using and I can make sure you're added to the [SuperCreate](https://addons-server.readthedocs.org/en/latest/topics/api/accounts.html#super-creation) so you can run the tests that involve logging in.

**Comment by davehunt** [comment]
@Edely you now have SuperCreate permission.

**Comment by Edely** [comment]
Thanks!

**Comment by Edely** [comment]
Hi, I having a little of trouble authenticating my user. 

First I created a jwt-token as explained here: https://addons-server.readthedocs.org/en/latest/topics/api/auth.html

I used my JWT issuer and JWT secret to generate the token using Node.js. Ok.

The problem is when I make the requisition I keep receiving this message:

`{"detail": "Unknown JWT iss (issuer)"}`

What I am typing is:

```
curl https://addons.mozilla.org/api/v3/accounts/profile/ \
     -H 'Authorization: JWT MY_TOKEN_COMES_HERE

```
I tried regenerate the API credentials, but I keep receiving this same message.

**Comment by davehunt** [comment]
These tests use pyjwt to [generate the token](https://github.com/mozilla/Addon-Tests/blob/016e4d39921abb8aa7062c0f233bdfee8d083f56/tests/conftest.py#L42-L49) are you able to take advantage of that existing code? I've not tried with node and curl, but it sounds like your token is missing the iss value?

**Comment by davehunt** [comment]
I just noticed your curl command points to addons.mozilla.org (prod), but I suspect your API credentials are for addons-dev.allizom.org (dev) or addons.allizom.org (staging)?

**Comment by Edely** [comment]
I decided to restart from beginning and now everything is working fine and I am able to run the tests with my API credentials. Thanks.

**Comment by rbillings** [comment]
Are you still working on this @Edely ?

**Comment by Edely** [comment]
Yes, I am. Sorry for the delay. I am going to send the code soon.

**Comment by davehunt** [comment]
Any updates on this @Edely or should we free this up for someone else to work on?

**Comment by Edely** [comment]
It is a good idea to pass to someone else. I will not be able to work on this. Thank you for the patience, anyway.

**Comment by davehunt** [comment]
No problem @Edely, if you change your mind just let us know. :)

**Comment by justinpotts** [comment]
@krupa @davehunt: Is this an instance where it would be a good idea to use [Project Axl](https://justinpotts.github.io/project-axl-frontend/)? If so, would using the frontend work? I have an idea on how to use a command line option, just not sure how we would integrate it with the test, without hosting the generator file inside the test repo itself.

**Comment by davehunt** [comment]
Fixed by #868
